File: data-skill/templates/r/plot_box.R
#!/usr/bin/env Rscript
# Box Plot Template
# Purpose: Create box plot for distribution comparison

library(tidyverse)

# ============================================================================
# USER INPUT: Modify these variables
# ============================================================================
data_file <- "data.csv"
numeric_var <- "numeric_variable"   # Change to your numeric variable
group_var <- "group_variable"       # Change to your grouping variable
# ============================================================================

# Load data
df <- read_csv(data_file)

# ============================================================================
# THEME
# ============================================================================
theme_report <- function() {
  theme_minimal(base_size = 11) +
    theme(
      plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
      axis.title = element_text(size = 11, face = "bold"),
      axis.text = element_text(size = 10, color = "black"),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(color = "gray90", linewidth = 0.3),
      panel.grid.minor = element_blank(),
      panel.border = element_rect(fill = NA, color = "black", linewidth = 0.5),
      plot.margin = margin(10, 10, 10, 10)
    )
}

# WHO color scheme
who_colors <- list(
  blue = "#4D9DE0",
  navy = "#104E8B",
  gray = "#A8A8A8"
)

# ============================================================================
# CREATE PLOT
# ============================================================================
p <- ggplot(df, aes(x = .data[[group_var]], y = .data[[numeric_var]])) +
  geom_boxplot(fill = who_colors$blue, alpha = 0.8) +
  geom_point(position = position_jitter(width = 0.2), alpha = 0.3, size = 1) +
  labs(
    title = "Box Plot Title",
    x = "Group Label",
    y = "Value Label"
  ) +
  theme_report()

# ============================================================================
# SAVE OUTPUT
# ============================================================================
ggsave(
  filename = "box_plot.png",
  plot = p,
  width = 8,
  height = 6,
  dpi = 300
)

cat("Plot saved to: box_plot.png\n")

File: data-skill/templates/python/plot_box.py
#!/usr/bin/env python3
# Box Plot Template
# Purpose: Create box plot for distribution comparison

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# ============================================================================
# USER INPUT: Modify these variables
# ============================================================================
data_file = "data.csv"
numeric_var = "numeric_variable"   # Change to your numeric variable
group_var = "group_variable"       # Change to your grouping variable
# ============================================================================

# Load data
df = pd.read_csv(data_file)

# ============================================================================
# THEME SETTINGS
# ============================================================================
sns.set_theme(style="whitegrid")
plt.rcParams['font.size'] = 11
plt.rcParams['axes.labelsize'] = 11
plt.rcParams['axes.titlesize'] = 14
plt.rcParams['axes.titleweight'] = 'bold'

# WHO color scheme
who_colors = {
    'blue': '#4D9DE0'
}

# ============================================================================
# CREATE PLOT
# ============================================================================
fig, ax = plt.subplots(figsize=(8, 6))

sns.boxplot(data=df, x=group_var, y=numeric_var,
            color=who_colors['blue'], ax=ax)

# Add jitter points
sns.stripplot(data=df, x=group_var, y=numeric_var,
              color='black', alpha=0.3, size=3, ax=ax)

ax.set_title("Box Plot Title")
ax.set_xlabel("Group Label")
ax.set_ylabel("Value Label")
ax.spines['top'].set_visible(True)
ax.spines['right'].set_visible(True)
ax.xaxis.grid(False)

plt.tight_layout()

# ============================================================================
# SAVE OUTPUT
# ============================================================================
plt.savefig("box_plot.png", dpi=300, bbox_inches='tight')
print("Plot saved to: box_plot.png")
plt.close()
